### PR TITLE
Fix: object-shorthand should only lint computed methods (fixes #6015)

### LINKED ISSUE #6015: Expected property shorthand warning on computed property
**What version of ESLint are you using?**
2.9.0

**What parser (default, Babel-ESLint, etc.) are you using?**
babel-eslint = 6.0.4

**Please show your full configuration:**
```
rules: {
   ...
   "object-shorthand": 1,
   ...
}
```
**What did you do? Please include the actual source code causing the issue.**
```
const constant = 'foo'
....
const obj = { [constant]: constant }
```

**What did you expect to happen?**
This should pass or I am missing something on how to do it.
**What actually happened? Please include the actual, raw output from ESLint.**
warning  Expected property shorthand  object-shorthand

### PR DESCRIPTION
**Pull request by kaicataldo**

### CODE REVIEW

**Comment by mysticatea** [comment]
LGTM :+1:

**Comment by afc163** [comment]
LGTM

**Comment by lemonmade** [comment]
Sorry for introducing this bug :(

File: lib/rules/object-shorthand.js [hunk 1]
```diff
@@ -109,7 +109,7 @@ module.exports = {
                 }
 
                 // only computed methods can fail the following checks
-                if (!APPLY_TO_METHODS && node.computed) {
+                if ((!APPLY_TO_METHODS || node.value.type !== "FunctionExpression") && node.computed) {
```
Comment (lemonmade) [inline lib/rules/object-shorthand.js]: I think you can drop the ``APPLY_TO_METHODS` check, the only computed case we care about is `node.value.type === "FunctionExpression" && node.computed`.

Comment (kaicataldo) [inline lib/rules/object-shorthand.js]: That makes sense - thanks!

**Comment by kaicataldo** [comment]
@lemonmade No need to apologize - it's part of the process of making things better

**Comment by kaicataldo** [comment]
Updated with @lemonmade's suggestion and added one more test

**Comment by nzakas** [comment]
Lgtm

**Pull request merged**